class.jetpack-import-stats.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+class Jetpack_Import_Stats {
+	static $known_importers = array(
+		'Blogger_Importer' => 'blogger',
+		'LJ_API_Import' => 'livejournal',
+		'MT_Import' => 'mt',
+		'RSS_Import' => 'rss',
+		'WC_Tax_Rate_Importer' => 'woo-tax-rate',
+		'WP_Import' => 'wordpress',
+	);
+
+	static $action_event_name_map = array(
+		'import_start' => 'jetpack_import_start',
+		'import_done'  => 'jetpack_import_done',
+		'import_end'   => 'jetpack_import_done',
+	);
+
+	public static function init() {
+		// Only handle import actions for sites that have agreed to TOS 
+		if ( Jetpack::jetpack_tos_agreed() ) {
+			add_action( 'import_start', array( 'Jetpack_Import_Stats', 'log_import_progress' ) );
+			add_action( 'import_done',  array( 'Jetpack_Import_Stats', 'log_import_progress' ) );
+			add_action( 'import_end',   array( 'Jetpack_Import_Stats', 'log_import_progress' ) );
+		}
+	}
+
+	private static function get_calling_class() {
```

Comment:
It might be worth sharing this code that detects import type with the import sync events (https://github.com/Automattic/jetpack/blob/e59cb3789aee320bfa157fefc5d976bf31757001/sync/class.jetpack-sync-module-import.php), since it looks like it's solving the same problem.